control_protocol.md

```diff
@@ -0,0 +1,293 @@
+    CA ->> Co1: CB|Co1.CA| Give me property A.
+:::
+
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> Co1: Co2.CB|Co1.CA| Give me property A.
+    Note over Co1,Co2: Co1.DEALER socket sends to Co2.ROUTER
+    Co1 ->> Co2: Co2.CB|Co1.CA| Give me property A.
+    Co2 ->> CB: Co2.CB|Co1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> Co2: Co1.CA|Co2.CB| Property A has value 5.
+    Note over Co1,Co2: Co2.DEALER socket sends to Co1.ROUTER
+    Co2 ->> Co1: Co1.CA|Co2.CB| Property A has value 5.
+    Co1 ->> CA: Co1.CA|Co2.CB| Property A has value 5.
+:::
+
+Prerequisite of Communication between two Components are:
+- Both Components are connected to a Coordinator and {ref}`signed in<sign-in>`.
+- Both Components are either connected to the same Coordinator (example one), or their Coordinators are connected to each other (example two).
+
+
+The following flow chart shows the decision scheme and message modification in a Coordinator.
+`Node0`, `NodeR` are placeholders for sender and recipient Node ID.
+`Recipient` is a placeholder for the recipient Component ID.
+`IA` is the connection identity of `Co1.CA` or of `Node0.COORDINATOR`.
+`IB` is the connection identities of `Co1.Recipient`.
+Bold arrows indicate message flow, thin lines indicate decision flow.
+Thin, dotted lines indicate decision flow in case of errors
+
+:::{mermaid}
+flowchart TB
+    C([Node0.CA DEALER]) == "NodeR.Recipient|Node0.CA|Content" ==> R0
+    R0[receive] == "IA|NodeR.Recipient|Node0.CA|Content" ==> CN0{Node0 == Co1}
+    CN0-->|no| RemIdent
+    CN0-->|yes| Clocal{CA in <br>local address book?}
+    Clocal -->|yes| CidKnown{IA is CA's identity<br> in address book?}
+    CidKnown -->|yes| RemIdent
+    Clocal -.->|no| E1[ERROR: Sender did not sign in] ==>|"IA|Node0.CA|Co1.COORDINATOR|ERROR: Sender dit not sign in"| S
+    S[send] ==> WA([Node0.CA DEALER])
+    CidKnown -.->|no| E2[ERROR: ID and identity do not match]==>|"IA|Node0.CA|Co1.COORDINATOR|ERROR: ID and identity do not match"| S
+    RemIdent[remove sender identity] == "NodeR.Recipient|Node0.CA|Content" ==> CNR
+    CNR -- "is None" --> Local
+    CNR{NodeR} -- "== Co1"--> Local
+    Local{Recipient<br>==<br>COORDINATOR} -- "yes" --> Self([Message for Co1<br> itself])
+    Local -- "no" --> Local2a{Recipient in Address book}
+    Local2a -->|yes, with Identity IB| Local2
+    Local2[add Recipient identity IB] == "IB|NodeR.Recipient|Node0.CA|Content" ==> R1[send]
+    R1 == "NodeR.Recipient|Node0.CA|Content" ==> W1([Wire to Co1.Recipient DEALER])
+    Local2a -.->|no| E3[ERROR Recipient unknown<br>send Error to original sender] ==>|"Node0.CA|Co1.COORDINATOR|ERROR Co1.Recipient is unknown"|CNR
+    CNR -- "== connected Coordinator Co2" --> Keep
+    Keep[send to Coordinator Co2] == "NodeR.Recipient|Node0.CA|Content" ==> R2[send]
+    R2 == "NodeR.Recipient|Node0.CA|Content" ==> W2([Wire to Co2 ROUTER])
+    CNR -- "reachable via Coordinator CoX" --> Augment
```

Comment:
Should we leave out the advanced routing (more than two Coordinators) for now, and just add it in a remark/todo/issue? It could help keeping focus (and the diagrams smaller :D), and for what we imagine so far I don't think multi-hop is essential?

Reply:
Ok